cpp/ql/src/semmle/code/cpp/AutogeneratedFile.qll

```diff
@@ -36,12 +112,13 @@ predicate hasPragmaDifferentFile(File f) {
  */
 class AutogeneratedFile extends File {
   cached AutogeneratedFile() {
-    exists(int limit, int head |
-      head <= 5 and
-      limit = max(int line | locations_default(_, underlyingElement(this), head, _, line, _)) + 5
-    |
-      exists (Comment c | c.getFile() = this and c.getLocation().getStartLine() <= limit and isAutogeneratedComment(c))
-    )
-    or hasPragmaDifferentFile(this)
+    autogeneratedComment(
+      concat(Comment c |
```

Comment:
I got some speed-up out of changing `concat` to `strictconcat`. In general it's always faster to use the strict aggregates.

Reply:
Good idea, I'd be surprised if it makes a big difference in this case but it shouldn't do any harm!

Reply:
It does, in fact, appear to make quite a big difference (done).